### DISCUSSION TITLE
RoboRock S8 - HA assistant

### DISCUSSION THREAD

**Discussion by arwindersingh82** [humbertogontijo/homeassistant-roborock#665] [General]
I've just bought a new RoboRock S8 - I am currently trying to move away from having to use the app for all the functions.  I basically have set a custom setting for each room, is there a way to replicate this in HA - or don't i need to do this because it will automatically use this?

Basically, i have set how the robot should handle each room as custom setting - i.e
kitchen - repeat 2, vac then mop, intensive mop
living room - vac only - repeat 1, 
bathroom - repeat 2, intensive mop 
etc..

I have then created a set of routines to use this- 
- Clean upstairs bathrooms
- Clean all upstairs 
- Clean Kitchen
- Clean downstairs
- etc.

I ideally wanted to move away from these routines and just create jobs in node red to clean kitch, clean living room etc.  I can do the basic and i have a tested a flow which i think will clean the kitchen, utility and hallway.  So i can create all those.

How do i set mop intensity or other settings - or is it that because i have setup each room in the app, when i run the job in api it will use that and hoover based on that?

`{
    "command": "app_segment_clean",
    "params": [
        {
            "segments": [
                17,
                18,
                19
            ],
            "repeat": 2
        }
    ]
}`

I created that to clean the hallway, living room and utility - is that all i need to do or do i need to enhance this to change intensity for each room etc.

**Comment by UsefulVid** [comment]
I was able to do this in the device:
![image](https://github.com/user-attachments/assets/1fa1d56f-8e09-4b1c-ba6c-3f37fa89089a)
